docs/adr/0002-delivery-channel-creation-and-integration.md

```diff
@@ -0,0 +1,102 @@
+
+### Management Api and usable Delivery Channels
+
+The management API will host configuration per tenant for what communication types can be sent through which delivery channels in the following format:
+
+```json
+[
+  {
+    "id": "dc1",
+    "displayName": "Airship",
+    "communicationTypes": ["email", "sms", "web-push"]
+  }, 
+  {
+    "id": "dc2",
+    "displayName": "Twilio",
+    "communicationTypes": ["sms"]
+  }
+]
+```
+where 'communicationTypes' specifies what communication types are supported in this Delivery Channel. This allows for consumers of Marain.UserNotifications to specify if they want to only use one kind communication type from a certain Delivery Channel that supports multiple communication types, eg: an organisation using `Marain.UserNotifications` can specify that they want to use sms's from twilio and only communication types from that delivery channel.
+
+A delivery channel will have a generic api that can be consumed by the Management Api and be able to transform data provided in the Notification into a format that is specific to that communication type (this will be discussed in a sequential adr). The result of this call will be a callback uri/null depending on how the delivery channel has been implemented.
+
+### Delivery Channel configuration 
+
+A delivery channel will also need the configuration of the underlying platform to be stored somewhere (api keys, secrets, etc). This will be stored in the `Marain.UserNotifications` keyvault. If a delivery channel needs to be used, there are two things that need to be added/modified:
+1. Modify the `Marain.UserNotifications` key vault secrets setion with the correct key/value pair (this is described in more details below).
+2. Update the configuration for the delivery channel in the management api to specify which communication types will use the delivery channel. 
+
+Keys for the different platforms in the delivery channels will be stored in a format as follows:
+`<TenantId>-<key_name>`
+ an example with this format would look like:
+`75b9261673c2714681f14c97bc0439fb00000000000000000000000000000000-ApiMasterSecret`
+
+A key vault will be associated to each channel in the `Marain.UserNotifications` These key vaults will be created with the following format:
+`deliverychannel-<ChannelName>`
+an example with this format would be:
+`deliverychannel-airship`
+
+The Delivery Channel will also have to have it's own kind of configuration specifying the supported Communication Types. These will be hardcoded values and created when the coding of the new delivery channel is done. 
```

Comment:
Ah - there's some overlap with my previous comment here...

Is the config you're talking about in this section separate from that discussed in the previous one? If so, why? Aren't they fundamentally intertwined?

Reply:
Yes, I've removed the old section.